Patch Tuesday for July 2024

BillyONeal opened the pull request:
* Add Azure CLI to all the machines
* Update PowerShell
* Update Visual Studio
* Remove no longer used Android NDK
* Get rid of the mintsas job and Managed Identity in favor of Workload Identity Federation which saves a job and some cores.

BillyONeal committed:
Add Azure CLI.

BillyONeal committed:
Update PowerShell.

BillyONeal committed:
Update Visual Studio

BillyONeal committed:
Remove android-ndk-r25c

BillyONeal committed:
Update android pool.

BillyONeal committed:
Update Windows pool to 2024-07-11

BillyONeal committed:
Update macOS pool.

BillyONeal committed:
Get rid of the mintsas job and Managed Identity in favor of Workload Identity Federation

BillyONeal committed:
Add Southeast Asia replica to Windows.

BillyONeal committed:
Add missing SAS question mark

BillyONeal committed:
Fix target region typo.

BillyONeal committed:
Update Windows pool to VMs with Azure CLI fixed.

BillyONeal commented:
Looks like the arm64-osx regressions are caused by the version of CMake changing:

REGRESSION: arg-router:arm64-osx failed with BUILD_FAILED. If expected, add arg-router:arm64-osx=fail to /Users/vcpkg/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

```
[1/1] "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/arg-router/src/v1.4.0-a16f5c5738.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/arg-router_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DINSTALLATION_ONLY=ON" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
FAILED: ../CMakeCache.txt 
"/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/arg-router/src/v1.4.0-a16f5c5738.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/arg-router_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DINSTALLATION_ONLY=ON" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.2s)
CMake Error at cmake/build_types/library.cmake:6 (target_link_libraries):
  The link interface of target "arg_router" contains:

    Boost::boost

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  CMakeLists.txt:159 (include)


-- Generating done (0.0s)
```

REGRESSION: libpqxx:arm64-osx failed with BUILD_FAILED. If expected, add libpqxx:arm64-osx=fail to /Users/vcpkg/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

```
"/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir "../../arm64-osx-dbg" "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/libpqxx/src/7.9.0-eff3c4d31f.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/libpqxx_arm64-osx/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DSKIP_BUILD_TEST=ON" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at cmake/config.cmake:14 (include):
  include could not find requested file:

    CMakeDetermineCompileFeatures
Call Stack (most recent call first):
  CMakeLists.txt:27 (include)


-- Found PostgreSQL: /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libpq.a (found version "16.2")
-- Found OpenSSL: /Users/vcpkg/Data/installed/arm64-osx/debug/lib/libcrypto.a (found version "3.3.1")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for poll
-- Looking for poll - found
CMake Error at cmake/config.cmake:42 (cmake_determine_compile_features):
  Unknown CMake command "cmake_determine_compile_features".
Call Stack (most recent call first):
  CMakeLists.txt:27 (include)
```

REGRESSION: qtwebchannel:arm64-osx failed with BUILD_FAILED. If expected, add qtwebchannel:arm64-osx=fail to /Users/vcpkg/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

```
[1/2] "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/qtwebchannel/src/here-src-6-0922b9c02f.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/qtwebchannel_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DQT_NO_FORCE_SET_CMAKE_BUILD_TYPE:BOOL=ON" "-DQT_USE_DEFAULT_CMAKE_OPTIMIZATION_FLAGS:BOOL=ON" "-DCMAKE_FIND_PACKAGE_TARGETS_GLOBAL=ON" "-DQT_BUILD_EXAMPLES:BOOL=OFF" "-DQT_BUILD_TESTS:BOOL=OFF" "-DQT_BUILD_BENCHMARKS:BOOL=OFF" "-DHOST_PERL:PATH=/usr/bin/perl" "-DINSTALL_BINDIR:STRING=bin" "-DINSTALL_LIBEXECDIR:STRING=bin" "-DINSTALL_PLUGINSDIR:STRING=Qt6/plugins" "-DINSTALL_QMLDIR:STRING=Qt6/qml" "-DCMAKE_REQUIRE_FIND_PACKAGE_Qt6Quick=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Qt6Quick=ON" "-DQT_SYNCQT:PATH=/Users/vcpkg/Data/installed/arm64-osx/tools/Qt6/bin/syncqt.pl" "-DQT_NO_FORCE_SET_CMAKE_BUILD_TYPE:BOOL=ON" "-DQT_MKSPECS_DIR:PATH=/Users/vcpkg/Data/installed/arm64-osx/share/Qt6/mkspecs" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DINSTALL_DOCDIR:STRING=doc/Qt6/" "-DINSTALL_INCLUDEDIR:STRING=include/Qt6/" "-DINSTALL_DESCRIPTIONSDIR:STRING=share/Qt6/modules" "-DINSTALL_MKSPECSDIR:STRING=share/Qt6/mkspecs" "-DINSTALL_TRANSLATIONSDIR:STRING=translations/Qt6/"
FAILED: ../CMakeCache.txt 
"/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/qtwebchannel/src/here-src-6-0922b9c02f.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/qtwebchannel_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DQT_NO_FORCE_SET_CMAKE_BUILD_TYPE:BOOL=ON" "-DQT_USE_DEFAULT_CMAKE_OPTIMIZATION_FLAGS:BOOL=ON" "-DCMAKE_FIND_PACKAGE_TARGETS_GLOBAL=ON" "-DQT_BUILD_EXAMPLES:BOOL=OFF" "-DQT_BUILD_TESTS:BOOL=OFF" "-DQT_BUILD_BENCHMARKS:BOOL=OFF" "-DHOST_PERL:PATH=/usr/bin/perl" "-DINSTALL_BINDIR:STRING=bin" "-DINSTALL_LIBEXECDIR:STRING=bin" "-DINSTALL_PLUGINSDIR:STRING=Qt6/plugins" "-DINSTALL_QMLDIR:STRING=Qt6/qml" "-DCMAKE_REQUIRE_FIND_PACKAGE_Qt6Quick=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Qt6Quick=ON" "-DQT_SYNCQT:PATH=/Users/vcpkg/Data/installed/arm64-osx/tools/Qt6/bin/syncqt.pl" "-DQT_NO_FORCE_SET_CMAKE_BUILD_TYPE:BOOL=ON" "-DQT_MKSPECS_DIR:PATH=/Users/vcpkg/Data/installed/arm64-osx/share/Qt6/mkspecs" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64" "-DINSTALL_DOCDIR:STRING=doc/Qt6/" "-DINSTALL_INCLUDEDIR:STRING=include/Qt6/" "-DINSTALL_DESCRIPTIONSDIR:STRING=share/Qt6/modules" "-DINSTALL_MKSPECSDIR:STRING=share/Qt6/mkspecs" "-DINSTALL_TRANSLATIONSDIR:STRING=translations/Qt6/"
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- The C compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found ZLIB: optimized;/Users/vcpkg/Data/installed/arm64-osx/lib/libz.a;debug;/Users/vcpkg/Data/installed/arm64-osx/debug/lib/libz.a (found version "1.3.1")
-- Found WrapSystemZLIB: TRUE (found version "1.3.1")
-- Using system ZLIB.
-- Found WrapZLIB: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found WrapSystemDoubleConversion: /Users/vcpkg/Data/installed/arm64-osx/share/double-conversion/double-conversionConfig.cmake (found version "3.3.0")
-- Found Libb2: /Users/vcpkg/Data/installed/arm64-osx/lib/libb2.a
-- Found ICU: /Users/vcpkg/Data/installed/arm64-osx/include (found suitable version "74.2", minimum required is "50.1") found components: i18n uc data
-- Found WrapSystemPCRE2: TRUE (found version "10.43.0")
-- Using system PCRE2.
-- Found WrapPCRE2: TRUE
-- Found WrapZSTD: 1.5.6 (found suitable version "1.5.6", minimum required is "1.3")
-- Performing Test HAVE_GETTIME
-- Performing Test HAVE_GETTIME - Success
-- Performing Test HAVE_SHM_OPEN_SHM_UNLINK
-- Performing Test HAVE_SHM_OPEN_SHM_UNLINK - Success
-- Found WrapRt: TRUE
CMake Warning at /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/Qt6Config.cmake:226 (message):
  Failed to find optional Qt component "Quick".

  Expected Config file at
  "/Users/vcpkg/Data/installed/arm64-osx/share/Qt6Quick/Qt6QuickConfig.cmake"
  exists

Call Stack (most recent call first):
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  CMakeLists.txt:20 (find_package)


-- Found OpenSSL: /Users/vcpkg/Data/installed/arm64-osx/lib/libcrypto.a (found version "3.3.1")
-- Found WrapOpenSSLHeaders: /Users/vcpkg/Data/installed/arm64-osx/include (found version "3.3.1")
-- Found WrapOpenSSL: /Users/vcpkg/Data/installed/arm64-osx/lib/libcrypto.a (found version "3.3.1")
CMake Warning at /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package):
  Found package configuration file:

    /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickControlsTestUtilsPrivate/Qt6QuickControlsTestUtilsPrivateConfig.cmake

  but it set Qt6QuickControlsTestUtilsPrivate_FOUND to FALSE so package
  "Qt6QuickControlsTestUtilsPrivate" is considered to be NOT FOUND.  Reason
  given by package:

  Qt6QuickControlsTestUtilsPrivate could not be found because dependency
  Qt6Quick could not be found.

  Configuring with --debug-find-pkg=Qt6Quick might reveal details why the
  package was not found.

  Configuring with -DQT_DEBUG_FIND_PACKAGE=ON will print the values of some
  of the path variables that find_package uses to try and find the package.

Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicDependencyHelpers.cmake:111 (find_dependency)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/QmlPlugins/Qt6QuickControlsTestUtilsPrivatepluginDependencies.cmake:18 (_qt_internal_find_qt_dependencies)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/QmlPlugins/Qt6QuickControlsTestUtilsPrivatepluginConfig.cmake:51 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlPlugins.cmake:47 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlConfig.cmake:139 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicDependencyHelpers.cmake:111 (find_dependency)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestDependencies.cmake:42 (_qt_internal_find_qt_dependencies)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestConfig.cmake:43 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/Qt6Config.cmake:169 (find_package)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  CMakeLists.txt:20 (find_package)
```

REGRESSION: wg21-linear-algebra:arm64-osx failed with BUILD_FAILED. If expected, add wg21-linear-algebra:arm64-osx=fail to /Users/vcpkg/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

```
[1/2] "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/wg21-linear-algebra/src/v0.7.3-fbdc4bfd46.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/wg21-linear-algebra_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DLA_INSTALL=ON" "-DLA_BUILD_PACKAGE=OFF" "-DLA_ENABLE_TESTS=OFF" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
FAILED: ../CMakeCache.txt 
"/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/wg21-linear-algebra/src/v0.7.3-fbdc4bfd46.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/wg21-linear-algebra_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DLA_INSTALL=ON" "-DLA_BUILD_PACKAGE=OFF" "-DLA_ENABLE_TESTS=OFF" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/FetchContent.cmake:2111 (message):
  FETCHCONTENT_FULLY_DISCONNECTED is set to true, which requires the source
  directory for dependency mdspan to already be populated.  This generally
  means it must not be set to true the first time CMake is run in a build
  directory.  The following source directory should already be populated, but
  it doesn't exist:

    /Users/vcpkg/Data/b/wg21-linear-algebra/arm64-osx-rel/_deps/mdspan-src

  Policy CMP0170 controls enforcement of this requirement.
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
  CMakeLists.txt:25 (FetchContent_MakeAvailable)


-- Configuring done (0.6s)
CMake Error at CMakeLists.txt:92 (target_link_libraries):
  The link interface of target "wg21_linear_algebra" contains:

    std::mdspan

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

REGRESSION: ztd-text:arm64-osx failed with BUILD_FAILED. If expected, add ztd-text:arm64-osx=fail to /Users/vcpkg/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

```
[1/1] "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/ztd-text/src/d49f345060-dc09e6b2b0.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/ztd-text_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
FAILED: ../CMakeCache.txt 
"/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" -E chdir ".." "/opt/homebrew/Cellar/cmake/3.30.0/bin/cmake" "/Users/vcpkg/Data/b/ztd-text/src/d49f345060-dc09e6b2b0.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=/Users/vcpkg/Data/p/ztd-text_arm64-osx" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DCMAKE_MAKE_PROGRAM=/opt/homebrew/bin/ninja" "-DCMAKE_SYSTEM_NAME=Darwin" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/toolchains/osx.cmake" "-DVCPKG_TARGET_TRIPLET=arm64-osx" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=external" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=/Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=arm64" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=/Users/vcpkg/Data/work/2/s" "-D_VCPKG_INSTALLED_DIR=/Users/vcpkg/Data/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_OSX_ARCHITECTURES=arm64"
CMake Warning at /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/FetchContent.cmake:2111 (message):
  FETCHCONTENT_FULLY_DISCONNECTED is set to true, which requires the source
  directory for dependency ztd.cmake to already be populated.  This generally
  means it must not be set to true the first time CMake is run in a build
  directory.  The following source directory should already be populated, but
  it doesn't exist:

    /Users/vcpkg/Data/b/ztd-text/arm64-osx-rel/_deps/ztd.cmake-src

  Policy CMP0170 controls enforcement of this requirement.
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
  CMakeLists.txt:42 (FetchContent_MakeAvailable)


-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.4s)
CMake Error at CMakeLists.txt:169 (target_link_libraries):
  The link interface of target "ztd.text" contains:

    ztd::cuneicode

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    VCPKG_PLATFORM_TOOLSET
    VCPKG_SET_CHARSET_FLAG
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
ninja: build stopped: subcommand failed.
```

BillyONeal commented:
`x64-osx` died somehow, I'm going to assume arm64-osx is a reasonable proxy for what a full build will say given that we didn't make major changes:

```
2024-07-14T02:25:15.6934120Z Installing 1164/2195 kf5texteditor:x64-osx@5.98.0...
2024-07-14T02:25:15.6935090Z Building kf5texteditor:x64-osx@5.98.0...
2024-07-14T02:25:15.7420940Z -- Downloading https://github.com/KDE/ktexteditor/archive/v5.98.0.tar.gz -> KDE-ktexteditor-v5.98.0.tar.gz...
2024-07-14T02:25:16.7031750Z -- Extracting source /Users/vcpkg/Data/downloads/KDE-ktexteditor-v5.98.0.tar.gz
2024-07-14T02:25:17.4460630Z -- Using source at /Users/vcpkg/Data/b/kf5texteditor/src/v5.98.0-94e36b3867.clean
2024-07-14T02:25:17.4791600Z -- Found external ninja('1.12.1').
2024-07-14T02:25:17.4843240Z -- Configuring x64-osx
2024-07-14T02:25:36.0791600Z -- Building x64-osx-dbg
2024-07-15T07:33:00.7590090Z ##[error]The Operation will be canceled. The next steps may not contain expected logs.
2024-07-15T07:33:00.8332220Z ##[error]The operation was canceled.
2024-07-15T07:33:00.8335610Z ##[section]Finishing: *** Test Modified Ports
```

BillyONeal committed:
[arg-router] Always depend on Boost.

Submitted upstream as https://github.com/cmannett85/arg_router/pull/398

BillyONeal committed:
[wg21-linear-algebra] Use external mdspan

BillyONeal committed:
[ztd-cuneicode,ztd-text] Fix referring to nonexistent target and out of source contents.

BillyONeal commented:
Ah, the actual qt errors are not what I expected:

```console
CMake Error at /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQmlInspectorServiceFactoryPluginTargets.cmake:60 (set_target_properties):
  The link interface of target "Qt6::QQmlInspectorServiceFactoryPlugin"
  contains:

    Qt6::QuickPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQmlInspectorServiceFactoryPluginConfig.cmake:54 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicPluginHelpers.cmake:518 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlPlugins.cmake:75 (__qt_internal_include_plugin_packages)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlConfig.cmake:139 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicDependencyHelpers.cmake:111 (find_dependency)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestDependencies.cmake:42 (_qt_internal_find_qt_dependencies)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestConfig.cmake:43 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/Qt6Config.cmake:169 (find_package)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  CMakeLists.txt:20 (find_package)


CMake Error at /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQmlPreviewServiceFactoryPluginTargets.cmake:60 (set_target_properties):
  The link interface of target "Qt6::QQmlPreviewServiceFactoryPlugin"
  contains:

    Qt6::QuickPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQmlPreviewServiceFactoryPluginConfig.cmake:54 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicPluginHelpers.cmake:518 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlPlugins.cmake:75 (__qt_internal_include_plugin_packages)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlConfig.cmake:139 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicDependencyHelpers.cmake:111 (find_dependency)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestDependencies.cmake:42 (_qt_internal_find_qt_dependencies)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestConfig.cmake:43 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/Qt6Config.cmake:169 (find_package)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  CMakeLists.txt:20 (find_package)


CMake Error at /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQuickProfilerAdapterFactoryPluginTargets.cmake:60 (set_target_properties):
  The link interface of target "Qt6::QQuickProfilerAdapterFactoryPlugin"
  contains:

    Qt6::QuickPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QQuickProfilerAdapterFactoryPluginConfig.cmake:54 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicPluginHelpers.cmake:518 (include)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlPlugins.cmake:75 (__qt_internal_include_plugin_packages)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6Qml/Qt6QmlConfig.cmake:139 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /opt/homebrew/Cellar/cmake/3.30.0/share/cmake/Modules/CMakeFindDependencyMacro.cmake:76 (find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/QtPublicDependencyHelpers.cmake:111 (find_dependency)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestDependencies.cmake:42 (_qt_internal_find_qt_dependencies)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6QuickTest/Qt6QuickTestConfig.cmake:43 (include)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  /Users/vcpkg/Data/installed/arm64-osx/share/Qt6/Qt6Config.cmake:169 (find_package)
  /Users/vcpkg/Data/work/2/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  CMakeLists.txt:20 (find_package)
```

Neumann-A commented:
@BillyONeal: looks like a dep on qtdeclarative is missing?` Maybe the qml dep is not correctly controlled anymore.

Neumann-A commented:
@BillyONeal Try adding CMAKE_DISABLE_FIND_PACKAGE_Qt6Qml to the feature check

BillyONeal committed:
[qtwebchannel] Fix qml installed but disabled builds.

Co-authored by: Alexander Neumann <30894796+Neumann-A@users.noreply.github.com>

BillyONeal commented:
libpqxx fixed by https://github.com/microsoft/vcpkg/pull/39927

BillyONeal commented on ports/ztd-cuneicode/fix-cmake-install.patch:
```diff
@@ -91,5 +91,8 @@ index 4bfb041..823d50b 100644
  	get_target_property(ZTD_CUNEICODE_INCLUDE_DIRS
- 		ztd.cuneicode.single INTERFACE_INCLUDE_DIRECTORIES)
+-		ztd.cuneicode.single INTERFACE_INCLUDE_DIRECTORIES)
++		ztd.cuneicode INTERFACE_INCLUDE_DIRECTORIES)
```
Upstream merged this change: https://github.com/soasis/cuneicode/pull/12

BillyONeal commented on ports/ztd-text/portfile.cmake:
```diff
@@ -1,16 +1,29 @@
-  SOURCE_PATH ${SOURCE_PATH}
+  SOURCE_PATH "${SOURCE_PATH}"
+  OPTIONS
+    # See https://github.com/soasis/cmake/blob/c29df2f0b006f8b24214ccea0a7e2f8fbbe135ce/CMakeLists.txt#L43
+    "-DZTD_CMAKE_PACKAGES=${ZTD_CMAKE_SOURCE_PATH}/Packages"
+    "-DZTD_CMAKE_MODULES=${ZTD_CMAKE_SOURCE_PATH}/Modules"
+    "-DZTD_CMAKE_PROJECT_PRELUDE=${ZTD_CMAKE_SOURCE_PATH}/Includes/Project.cmake"
```
/cc @ThePhD I wasn't sure if this was the right way to devendor the dependency

ThePhD replied:
This looks fine enough to me!

dg0yt commented:
> REGRESSION: libpqxx:arm64-osx failed with BUILD_FAILED.

Fixed by https://github.com/microsoft/vcpkg/pull/39927

BillyONeal commented:
/azp run

BillyONeal commented:
/azp run
